src/servers/src/otlp/trace.rs

```diff
@@ -0,0 +1,397 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use api::v1::value::ValueData;
+use api::v1::{ColumnDataType, RowInsertRequests, Value};
+use common_grpc::writer::Precision;
+use common_time::time::Time;
+use itertools::Itertools;
+use opentelemetry_proto::tonic::collector::trace::v1::ExportTraceServiceRequest;
+use opentelemetry_proto::tonic::common::v1::any_value::Value as OtlpValue;
+use opentelemetry_proto::tonic::common::v1::{
+    AnyValue, ArrayValue, InstrumentationScope, KeyValue, KeyValueList,
+};
+use opentelemetry_proto::tonic::trace::v1::span::{Event, Link};
+use opentelemetry_proto::tonic::trace::v1::{Span, Status};
+use serde_json::json;
+
+use super::{GREPTIME_TIMESTAMP, GREPTIME_VALUE};
+use crate::error::Result;
+use crate::row_writer::{self, MultiTableData, TableData};
+
+const APPROXIMATE_COLUMN_COUNT: usize = 16;
+const TRACE_TABLE_NAME: &str = "traces_preview_v01";
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+struct TraceSpan {
+    // the following are tags
+    trace_id: String,
+    span_id: String,
+    parent_span_id: String,
+
+    // the following are fields
+    resource_attributes: String, // Map in the future
+    scope_name: String,
+    scope_version: String,
+    scope_attributes: String, // Map in the future
+    trace_state: String,
+    span_name: String,
+    span_kind: String,
+    span_status_code: String,
+    span_status_message: String,
+    span_attributes: String, // Map in the future
+    span_events: String,     // List in the future
+    span_links: String,      // List in the future
+    start: u64,              // in nanosecond
+    end: u64,                // in nanosecend
+
+    ts: i64, // timestamp
```

Comment:
We can use `start` as time index and do not need to duplicate it as a new field (in memory or at rest)